opsmngr/backup_administrator.go

```diff
@@ -0,0 +1,874 @@
+// Copyright 2019 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	backupAdministratorBlockstoreBasePath                    = "admin/backup/snapshot/mongoConfigs"
+	backupAdministratorFileSystemStoreConfigurationsBasePath = "admin/backup/snapshot/fileSystemConfigs"
+	backupAdministratorS3BlockstoreBasePath                  = "admin/backup/snapshot/s3Configs"
+	backupAdministratorOplogBasePath                         = "admin/backup/oplog/mongoConfigs"
+	backupAdministratorSyncBasePath                          = "admin/backup/sync/mongoConfigs"
+	backupAdministratorDaemonBasePath                        = "admin/backup/daemon/configs"
+	backupAdministratorProjectJobBasePath                    = "admin/backup/groups"
+)
+
+// BackupAdministratorService is an interface for using the Backup Administrator
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/nav/administration-backup/
+type BackupAdministratorService interface {
+	BlockstoreService
+	FileSystemStoreService
+	S3BlockstoreService
+	OplogService
+	SyncService
+	DaemonService
+	ProjectJobService
+}
+
+// BlockstoreService is an interface for using the Blockstore
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/blockstore-config/
+type BlockstoreService interface {
+	ListBlockstores(context.Context, *atlas.ListOptions) (*Blockstores, *atlas.Response, error)
+	GetBlockstore(context.Context, string) (*Blockstore, *atlas.Response, error)
+	CreateBlockstore(context.Context, *Blockstore) (*Blockstore, *atlas.Response, error)
+	UpdateBlockstore(context.Context, string, *Blockstore) (*Blockstore, *atlas.Response, error)
+	DeleteBlockstore(context.Context, string) (*atlas.Response, error)
+}
```

Comment:
Following my previous comment to follow what we've done so far this should be something like

```suggestion
// BlockstoreConfigService is an interface for using the Blockstore
// endpoints of the MongoDB Ops Manager API.
//
// See more: https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/blockstore-config/
type BlockstoreConfigService interface {
	List(context.Context, *atlas.ListOptions) (*Blockstores, *atlas.Response, error)
	Get(context.Context, string) (*Blockstore, *atlas.Response, error)
	Create(context.Context, *Blockstore) (*Blockstore, *atlas.Response, error)
	Update(context.Context, string, *Blockstore) (*Blockstore, *atlas.Response, error)
	Delete(context.Context, string) (*atlas.Response, error)
}
```

I would also keep them in separate files